src/System.ComponentModel.TypeConverter/src/System/Drawing/ImageConverter.cs

```diff
@@ -0,0 +1,77 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.ComponentModel;
+using System.Globalization;
+using System.IO;
+
+namespace System.Drawing
+{
+    public class ImageConverter : TypeConverter
+    {
+        public ImageConverter()
+        {
+        }
+
+        public override bool CanConvertFrom(ITypeDescriptorContext context, Type sourceType)
+        {
+            return (sourceType == typeof(byte[]));
+        }
+
+        public override bool CanConvertTo(ITypeDescriptorContext context, Type destinationType)
+        {
+            if ((destinationType == typeof(byte[])) || (destinationType == typeof(string)))
+            {
+                return true;
+            }
+            return false;
+        }
+
+        public override object ConvertFrom(ITypeDescriptorContext context, CultureInfo culture, object value)
+        {
+            byte[] bytes = value as byte[];
+            if (bytes == null)
+            {
+                return base.ConvertFrom(context, culture, value);
+            }
+
+            return Image.FromStream(new MemoryStream(bytes));
+        }
+
+        public override object ConvertTo(ITypeDescriptorContext context, CultureInfo culture, object value, Type destinationType)
+        {
+            if (destinationType == typeof(string))
+            {
+                if (value == null)
+                {
+                    return SR.none;
+                }
+                else if (value is Image)
+                {
+                    return value.ToString();
+                }
+            }
+            else if (destinationType == typeof(byte[]))
+            {
+                using (MemoryStream ms = new MemoryStream())
+                {
+                    ((Image)value).Save(ms, ((Image)value).RawFormat);
```

Comment:
This can throw NRE if `value` is `null`. I will fix it, but I am not sure how:

- Let it go to the end and `throw GetConvertFromException(value)`.
- Return `null`.
- Return empty `byte[]` array.

Suggestions?

Reply:
Based on https://referencesource.microsoft.com/#System.Drawing/commonui/System/Drawing/ImageConverter.cs,132:

```c#
return Array.Empty<byte>();
```

Reply:
Thanks. Fixed.